demos/chat_with_pdf.py

```diff
@@ -227,26 +250,24 @@ def __call__(self, file: str, chunk_size: int, chunking_technique: str) -> Any:
     def build_chain(self, vector_store):
         retriever = vector_store.as_retriever(search_kwargs={"k": self.top_k})
 
-        def format_docs(docs):
-            return "\n\n".join(doc.page_content for doc in docs)
-
-        prompt = ChatPromptTemplate.from_messages([
-            ("system", "You are a helpful AI assistant. Use the following pieces of context to answer the user's question. If you don't know the answer, just say that you don't know, don't try to make up an answer."),
-            ("system", "Context: {context}"),
-            ("human", "{question}"),
-            ("system", "Provide a helpful and accurate answer based on the given context and question:")
-        ])
-
-        rag_chain = (
-            {
-                "context": retriever | format_docs,
-                "question": RunnablePassthrough()
-            }
-            | prompt
-            | self.cached_llm
-            | StrOutputParser()
+        prompt = ChatPromptTemplate.from_messages(
+            [
+                (
+                    "system",
+                    "You are a helpful AI assistant. Use the following pieces of context to answer the user's question. If you don't know the answer, just say that you don't know, don't try to make up an answer.",
+                ),
+                ("system", "Context: {context}"),
+                ("human", "{input}"),
+                (
+                    "system",
+                    "Provide a helpful and accurate answer based on the given context and question:",
+                ),
+            ]
         )
 
+        combine_docs_chain = create_stuff_documents_chain(self.cached_llm, prompt)
```

Comment:
changed to this to get source_documents back for ragas. It could also be helpful for future stuff

Reply:
Got it, but I don't get the `stuff` in the name `create_stuff_documents_chain`?

Reply:
"stuff documents" is a type of LLM chain in LC from what I recall